### PR TITLE
E2E improvements

### PR DESCRIPTION
**Pull request by brunobar79**
Fixes APP-####

## What changed (plus any additional context for devs)
- Replace Hardhat in favor of Anvil
- Replace usage of `IS_TESTING` in favor of `IS_TEST`
- Fix the detox config to actually run the correct tests for serial & parallel. This wasn't working and tests were running on the wrong queue.
- Update the script with more efficient retry logic so it only runs what failed instead of the entire suite
- Run anvil separately so we can see the logs that are actually useful during the job execution

## Screen recordings / screenshots

Before ~ 21min

Screenshot 2024-12-21 at 8 02 51 PM

Now ~ 15min
Screenshot 2024-12-21 at 8 03 00 PM

## What to test
e2e should pass

### CODE REVIEW

File: e2e/helpers.ts [hunk 1]
```diff
@@ -467,16 +467,23 @@ export const getProvider: ProviderFunction = () => {
...
+  console.log('sending eth');
   await wallet.sendTransaction({
     to: TESTING_WALLET,
     value: parseEther('20'),
   });
+  console.log('sent eth');
   await delayTime('long');
+  console.log('checking balance');
   const balance = await provider.getBalance(TESTING_WALLET);
+  console.log('got balance', formatEther(balance));
```
Comment (brunobar79) [inline e2e/helpers.ts:486]: let's keep this log cause this part breaks often

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@bruno/switch-to-anvil/df5ee7a1ccbbeeaba357fe072558898162e52044.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@bruno/switch-to-anvil/df5ee7a1ccbbeeaba357fe072558898162e52044.ipa) for df5ee7a1ccbbeeaba357fe072558898162e52044

**Review by derHowie — APPROVED** [review]

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@bruno/switch-to-anvil/ee42ebc36cb1faf535c09546d5cbca05fe3f7dee.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@bruno/switch-to-anvil/ee42ebc36cb1faf535c09546d5cbca05fe3f7dee.ipa) for ee42ebc36cb1faf535c09546d5cbca05fe3f7dee

**Review by walmat — APPROVED** [review]

File: src/components/skeleton/Skeleton.tsx [hunk 2]
```diff
@@ -75,7 +75,7 @@ function Skeleton({
   skeletonColor?: string;
   width?: number;
 }) {
-  if (animated && IS_TESTING !== 'true') {
+  if (animated && IS_TEST) {
```
Comment (brunobar79) [inline src/components/skeleton/Skeleton.tsx:78]:
```suggestion
  if (animated && !IS_TEST) {
```

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@bruno/switch-to-anvil/316a81eab81035cd14f7d11c5869f74fc7e90c87.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@bruno/switch-to-anvil/316a81eab81035cd14f7d11c5869f74fc7e90c87.ipa) for 316a81eab81035cd14f7d11c5869f74fc7e90c87

**Pull request merged**